tools/deploy_ipk.sh

```diff
@@ -21,33 +27,35 @@ deploy() {
     DEST_FOLDER=/tmp/prplmesh_ipks
 
     echo "Removing previous ipks"
-    ssh "$TARGET" "rm -rf \"$DEST_FOLDER\" ; mkdir -p \"$DEST_FOLDER\""
+    eval ssh "$SSH_OPTIONS" "$TARGET" \""rm -rf $DEST_FOLDER ; mkdir -p $DEST_FOLDER\""
     echo "Copying $IPK to $TARGET:$DEST_FOLDER/$IPK_FILENAME"
-    scp "$IPK" "$TARGET:$DEST_FOLDER/$IPK_FILENAME"
+    eval scp "${SSH_OPTIONS} ${IPK} ${TARGET}:${DEST_FOLDER}/${IPK_FILENAME}"
 
-    ssh "$TARGET" <<EOF
+    eval ssh "$SSH_OPTIONS" "$TARGET" <<EOF
 # we don't want opkg to stay locked with a previous failed invocation.
 # when using this, make sure no one is using opkg in the meantime!
 pgrep opkg | xargs kill -SIGINT
 # remove any previously installed prplmesh:
 opkg remove prplmesh prplmesh-dwpal prplmesh-nl80211
 # currently opkg remove does not remove everything from /opt/prplmesh:
 rm -rf /opt/prplmesh
-opkg install -V2 "$DEST_FOLDER/$IPK_FILENAME"
+opkg install --force-reinstall -V2 "$DEST_FOLDER/$IPK_FILENAME"
```

Comment:
Are you really sure this is needed?
There is the 'opkg remove' right before, and the script can't get here if it failed (unless opkg remove doesn't report failures in its exit code?)

Reply:
I think I get it now - when I built prplmesh, it was in `bin/packages/mips_24kc_nomips16/iwlwav/prplmesh-dwpal_1.4.0-d78175d4-1.4.0_mips_24kc_nomips16.ipk` so `basename` would produce `prplmesh-dwpal_1.4.0-d78175d4-1.4.0_mips_24kc_nomips16.ipk` and not `prplmesh-dwpal` which is used to remove prplmesh in line 39 - I assume this is the reason that `opkg remove` didn't work for me. I wonder though how it works (if it works) in CI - perhaps it is different since it is built with SDK and not as part of the full firmware?

Reply:
So I was wrong. It is indeed `prplmesh-dwpal`, but removal fails none the less:
```
root@OpenWrt:~# opkg remove prplmesh-dwpal
No packages removed.
Collected errors:
 * print_dependents_warning: Package prplmesh-dwpal is depended upon by packages:
 * print_dependents_warning:    ltq-wlan-wave_6x-uci
 * print_dependents_warning: These might cease to work if package prplmesh-dwpal is removed.

 * print_dependents_warning: Force removal of this package with --force-depends.
 * print_dependents_warning: Force removal of this package and its dependents
 * print_dependents_warning: with --force-removal-of-dependent-packages.
root@OpenWrt:~#
```
So I guess that this we should definitely use either `--force-depends` for the `opkg remove` or `--force-reinstall` for the `opkg install`.

Reply:
OK so I fully understand this now - in CI, there is no dependencies on prplmesh since it is built using SDK, while on RAX40 latest firmware, prplmesh is already present and since it is selected by the wlan_wave_6x-uci package, opkg marks this as a dependency.

Reply:
Oh, right. I forgot about that. Either the fact that `ltq-wlan-wave_6x-uci` depends on prplMesh is a bug we should fix, or we should remove it as well when we remove prplMesh (probably `--force-removal-of-dependent-packages` does it).